nautobot/utilities/views.py

```diff
@@ -13,7 +17,19 @@
 #
 
 
-class ContentTypePermissionRequiredMixin(AccessMixin):
+class LoginMixin(AccessMixin):
+    def get_login_url(self):
+        login_url = super().get_login_url()
+
+        is_sso_auth_enabled = sso_auth_enabled(settings.AUTHENTICATION_BACKENDS)
+        backends = user_backends_data(self.request.user, settings.AUTHENTICATION_BACKENDS, Storage)["backends"]
+        if backends and is_sso_auth_enabled:
+            return reverse("social:begin", args=[backends[0]])
```

Comment:
Would it be possible to instead define `settings.LOGIN_URL` differently in the case where SSO auth is enabled? Or does this *have* to be done on the fly like this?

Reply:
I don't think its possible, for these reasons:

1. `LOGIN_URL` for SSO cant be hard-coded because it differs for different SSO auth backend, e.g. 
    i. `/login/saml` for `social_core.backends.saml.SAMLAuth`
    ii. `/login/google-auth2` for `social_core.backends.google.GoogleOAuth2` 
    iii. `/login/azuread-oauth2` for `social_core.backends.azuread.AzureADOAuth2` e.t.c

2. The `LOGIN_URL` cant be generated at `core.settings`/`nautobot_config` because ill need the `reverse(...)` function which cant be imported there because it trows an error